src/domain/eventGroups/utils.ts

```diff
@@ -0,0 +1,42 @@
+import { EventGroup_eventGroup_events_edges_node as EventNode } from '../../api/generatedTypes/EventGroup';
+import { Record } from '../../api/types';
+
+export enum EventGroupEventsPublishStatusEnum {
+  NotReady = 'notReady',
+  ReadyForFirstPublish = 'readyForFirstPublish',
+  ReadyForRepublish = 'readyForRepublish',
+  AllPublished = 'allPublished',
+}
+
+export const isEveryEventPublished = (events: EventNode[]) => {
+  return events?.every((event) => event.publishedAt !== null) || false;
+};
+
+export const isSomeEventsPublished = (events: EventNode[]) => {
+  return events?.some((event) => event.publishedAt !== null) || false;
+};
+
+export const isEveryEventReadyForPublish = (events: EventNode[]) => {
+  return events?.every((event) => event.readyForEventGroupPublishing) || false;
+};
+
+export const getEventGroupPublishStatus = (eventGroup: Record | undefined) => {
+  if (eventGroup && Boolean(eventGroup.publishedAt)) {
+    if (isEveryEventPublished(eventGroup.events)) {
+      return EventGroupEventsPublishStatusEnum.AllPublished;
+    }
+    if (
+      !isSomeEventsPublished(eventGroup.events) &&
+      isEveryEventReadyForPublish(eventGroup.events)
+    ) {
+      return EventGroupEventsPublishStatusEnum.ReadyForFirstPublish;
+    }
+    if (
+      isSomeEventsPublished(eventGroup.events) &&
+      isEveryEventReadyForPublish(eventGroup.events)
+    ) {
+      return EventGroupEventsPublishStatusEnum.ReadyForRepublish;
+    }
+  }
```

Comment:
I managed to spend a tiny bit too much time trying to debug my permissions on the backend, but finally noticed that the `Boolean(eventGroup.publishedAt)` in the first if-clause prevents showing the _publish_ button for the event group. Also I think we probably can remove the `isSomeEventsPublished` function like this:

 ```suggestion
  if (eventGroup) {
    if (isEveryEventPublished(eventGroup.events)) {
      return EventGroupEventsPublishStatusEnum.AllPublished;
    }
    else if (
      !Boolean(eventGroup.publishedAt) &&
      isEveryEventReadyForPublish(eventGroup.events)
    ) {
      return EventGroupEventsPublishStatusEnum.ReadyForFirstPublish;
    }
    else if (
      isEveryEventReadyForPublish(eventGroup.events)
    ) {
      return EventGroupEventsPublishStatusEnum.ReadyForRepublish;
    }
  }
```

Reply:
Took the suggested code